src/legacy/core_plugins/kibana/public/visualize/plugin.ts

```diff
@@ -159,19 +160,33 @@ export class VisualizePlugin implements Plugin {
   }
 
   public start(
-    { savedObjects: { client: savedObjectsClient } }: CoreStart,
+    core: CoreStart,
     { embeddables, navigation, data, share, visualizations }: VisualizePluginStartDependencies
   ) {
     this.startDependencies = {
       data,
       embeddables,
       navigation,
-      savedObjectsClient,
+      savedObjectsClient: core.savedObjects.client,
       share,
       visualizations,
     };
 
-    const embeddableFactory = new VisualizeEmbeddableFactory(visualizations.types);
-    embeddables.registerEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, embeddableFactory);
+    const savedVisualizations = createSavedVisLoader({
```

Comment:
@kertal Changes how this works in this PR and it looks like it will make this logic unnecessary: https://github.com/elastic/kibana/pull/54155

@ppisljar  Can you wait with your PR till that one is merged? This PR introduces `managementRegistry` and `uiModules` in this file which I would like to avoid because Matthias' PR will get rid of it anyway.

Reply:
seems that PR is reviewed and just waiting for the feature branch ?
i am fine with it merging first, this will probably anyway take another day to review

Reply:
if it happens the other way around (and this is ready first) i also don't think it should be a problem,  either @kertal removes the uiModules and managementRegistry from this file, or i can do it in followup

Reply:
Sounds good to me, please leave a `TODO` comment to document it somewhere.

Reply:
Better than a comment would be to pass in `uiModules` and `managementRegistry` as a `__LEGACY` plugin dependency, just to follow the convention of shimmed plugins.

Reply:
I've move #54155 to 7.7, it's no problem to wait until this is merged in